In [1]:
import pandas as pd
import numpy as np

from sklearn.preprocessing import StandardScaler, OneHotEncoder

In [2]:
df = pd.read_csv('C:/Users/reraveen/Jupyter/Dataset/mlt_retail_cust_eng_history.csv',dtype={'CUSTOMER_NO': object})


c:\users\reraveen\appdata\local\programs\python\python38\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (1,71) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
df.shape

(2137142, 74)

In [4]:
#The final check we want to do is see what data types we have
df.dtypes

CUSTOMER_NO        object
HOME_BRN           object
REP_CCY           float64
CUST_AGE            int64
MARITAL_STAT       object
                   ...   
PRED_CUST_LTV     float64
PRED_CUST_SEG     float64
PRED_RISK_TYPE     object
HIST_MOVE_DATE     object
PROD_RECO         float64
Length: 74, dtype: object

In [5]:
df[df.isnull().any(axis=1)]

CUSTOMER_NO HOME_BRN  REP_CCY  CUST_AGE MARITAL_STAT EMP_STAT  \
0         000000418        0      NaN        44            M        F   
1         000000422        0      NaN        39            M        F   
2         000000430        0      NaN        30            S        F   
3         000000433        0      NaN        77            M        U   
4         000000437        0      NaN        37            M        F   
...             ...      ...      ...       ...          ...      ...   
2137137   133146421      133      NaN        37            M        S   
2137138   091166504       91      NaN        34            S        F   
2137139   201166506      201      NaN        39            M        F   
2137140   000146441        0      NaN        22            S        F   
2137141   141166630      141      NaN        52            M        S   

        SPOUSE_EMP_STAT  NUM_DEPEND  CUST_INC  OVERALL_LIMIT  ...  \
0                     O           1     375.0            NaN  ...   
1                   NaN           2     506.0            NaN  ...   
2                   NaN           0       0.0            NaN  ...   
3                   NaN           0       0.0            NaN  ...   
4                     F           3     495.5            NaN  ...   
...                 ...         ...       ...            ...  ...   
2137137             NaN           3       0.0            NaN  ...   
2137138             NaN           1     200.0            NaN  ...   
2137139             NaN           3     304.0            NaN  ...   
2137140               O           0     145.0            NaN  ...   
2137141             NaN           9       0.0            NaN  ...   

         DATA_COLL_TO_DT  CHURN_STAT_DT  CHURN_STAT  PRED_CHURN_STAT  \
0              14-JAN-19      14-JAN-19           0              0.0   
1              14-JAN-19      14-JAN-19           0              0.0   
2              14-JAN-19      14-JAN-19           0              0.0   
3              14-JAN-19      14-JAN-19           1              1.0   
4              14-JAN-19      14-JAN-19           0              0.0   
...                  ...            ...         ...              ...   
2137137        31-DEC-17      31-MAR-18           0              NaN   
2137138        31-DEC-17      31-MAR-18           0              NaN   
2137139        31-DEC-17      31-MAR-18           0              NaN   
2137140        31-DEC-17      31-MAR-18           0              NaN   
2137141        31-DEC-17      31-MAR-18           0              NaN   

         PRED_CHURN_PROB  PRED_CUST_LTV  PRED_CUST_SEG  PRED_RISK_TYPE  \
0                    0.0          11.26            4.0         NO RISK   
1                    0.0         169.22            6.0         NO RISK   
2                    0.0         297.39            6.0         NO RISK   
3                    1.0            NaN            6.0            HIGH   
4                    0.0           0.00            6.0         NO RISK   
...                  ...            ...            ...             ...   
2137137              NaN            NaN            NaN             NaN   
2137138              NaN            NaN            NaN             NaN   
2137139              NaN            NaN            NaN             NaN   
2137140              NaN            NaN            NaN             NaN   
2137141              NaN            NaN            NaN             NaN   

         HIST_MOVE_DATE  PROD_RECO  
0             21-NOV-19        NaN  
1             21-NOV-19        NaN  
2             21-NOV-19        NaN  
3             21-NOV-19        NaN  
4             21-NOV-19        NaN  
...                 ...        ...  
2137137       16-APR-20        NaN  
2137138       16-APR-20        NaN  
2137139       16-APR-20        NaN  
2137140       16-APR-20        NaN  
2137141       16-APR-20        NaN  

[2137142 rows x 74 columns]

In [6]:
print(df['REP_CCY'].isnull().sum()) #since the number of NaN values equal to the Row Size we are removing the column

2137142


In [7]:
print(df['SPOUSE_EMP_STAT'].isnull().sum())

1967401


In [8]:
print(df['OVERALL_LIMIT'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

2137142


In [9]:

print(df['PRED_CHURN_STAT'].isnull().sum())#

1386551


In [10]:
#need to replace

In [11]:


print(df['PRED_CUST_LTV'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

1407325


In [12]:
#need to replace 

In [13]:

print(df['PRED_CUST_SEG'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

1386551


In [14]:
#need to replace
#from sklearn.impute import SimpleImputer
df.fillna(0, inplace=True)
df

CUSTOMER_NO HOME_BRN  REP_CCY  CUST_AGE MARITAL_STAT EMP_STAT  \
0         000000418        0      0.0        44            M        F   
1         000000422        0      0.0        39            M        F   
2         000000430        0      0.0        30            S        F   
3         000000433        0      0.0        77            M        U   
4         000000437        0      0.0        37            M        F   
...             ...      ...      ...       ...          ...      ...   
2137137   133146421      133      0.0        37            M        S   
2137138   091166504       91      0.0        34            S        F   
2137139   201166506      201      0.0        39            M        F   
2137140   000146441        0      0.0        22            S        F   
2137141   141166630      141      0.0        52            M        S   

        SPOUSE_EMP_STAT  NUM_DEPEND  CUST_INC  OVERALL_LIMIT  ...  \
0                     O           1     375.0            0.0  ...   
1                     0           2     506.0            0.0  ...   
2                     0           0       0.0            0.0  ...   
3                     0           0       0.0            0.0  ...   
4                     F           3     495.5            0.0  ...   
...                 ...         ...       ...            ...  ...   
2137137               0           3       0.0            0.0  ...   
2137138               0           1     200.0            0.0  ...   
2137139               0           3     304.0            0.0  ...   
2137140               O           0     145.0            0.0  ...   
2137141               0           9       0.0            0.0  ...   

         DATA_COLL_TO_DT  CHURN_STAT_DT  CHURN_STAT  PRED_CHURN_STAT  \
0              14-JAN-19      14-JAN-19           0              0.0   
1              14-JAN-19      14-JAN-19           0              0.0   
2              14-JAN-19      14-JAN-19           0              0.0   
3              14-JAN-19      14-JAN-19           1              1.0   
4              14-JAN-19      14-JAN-19           0              0.0   
...                  ...            ...         ...              ...   
2137137        31-DEC-17      31-MAR-18           0              0.0   
2137138        31-DEC-17      31-MAR-18           0              0.0   
2137139        31-DEC-17      31-MAR-18           0              0.0   
2137140        31-DEC-17      31-MAR-18           0              0.0   
2137141        31-DEC-17      31-MAR-18           0              0.0   

         PRED_CHURN_PROB  PRED_CUST_LTV  PRED_CUST_SEG  PRED_RISK_TYPE  \
0                    0.0          11.26            4.0         NO RISK   
1                    0.0         169.22            6.0         NO RISK   
2                    0.0         297.39            6.0         NO RISK   
3                    1.0           0.00            6.0            HIGH   
4                    0.0           0.00            6.0         NO RISK   
...                  ...            ...            ...             ...   
2137137              0.0           0.00            0.0               0   
2137138              0.0           0.00            0.0               0   
2137139              0.0           0.00            0.0               0   
2137140              0.0           0.00            0.0               0   
2137141              0.0           0.00            0.0               0   

         HIST_MOVE_DATE  PROD_RECO  
0             21-NOV-19        0.0  
1             21-NOV-19        0.0  
2             21-NOV-19        0.0  
3             21-NOV-19        0.0  
4             21-NOV-19        0.0  
...                 ...        ...  
2137137       16-APR-20        0.0  
2137138       16-APR-20        0.0  
2137139       16-APR-20        0.0  
2137140       16-APR-20        0.0  
2137141       16-APR-20        0.0  

[2137142 rows x 74 columns]

In [15]:


print(df['PRED_RISK_TYPE'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

0


In [16]:


print(df['PRED_RISK_TYPE'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

0


In [17]:

print(df['PROD_RECO'].isnull().sum())#since the number of NaN values equal to the Row Size we are removing the column

0


In [18]:
#object dtype columns
obj_df = df.select_dtypes(include=['object']).copy()
obj_df.head()

CUSTOMER_NO HOME_BRN MARITAL_STAT EMP_STAT SPOUSE_EMP_STAT RECORD_STAT  \
0   000000418        0            M        F               O           O   
1   000000422        0            M        F               0           O   
2   000000430        0            S        F               0           O   
3   000000433        0            M        U               0           O   
4   000000437        0            M        F               F           O   

  DATA_EXT_DT DATA_COLL_FROM_DT DATA_COLL_TO_DT CHURN_STAT_DT PRED_RISK_TYPE  \
0   15-JUL-19         14-JAN-18       14-JAN-19     14-JAN-19        NO RISK   
1   15-JUL-19         14-JAN-18       14-JAN-19     14-JAN-19        NO RISK   
2   15-JUL-19         14-JAN-18       14-JAN-19     14-JAN-19        NO RISK   
3   15-JUL-19         14-JAN-18       14-JAN-19     14-JAN-19           HIGH   
4   15-JUL-19         14-JAN-18       14-JAN-19     14-JAN-19        NO RISK   

  HIST_MOVE_DATE  
0      21-NOV-19  
1      21-NOV-19  
2      21-NOV-19  
3      21-NOV-19  
4      21-NOV-19

In [19]:
obj_df["HOME_BRN"].value_counts()

151    86488
31     60529
0      58316
153    53739
73     52832
       ...  
RET        3
146        3
RR1        3
111        3
PR1        3
Name: HOME_BRN, Length: 141, dtype: int64

since "HOME_BRN" column contains 141 val -so we need to segment it or 

In [20]:
obj_df["EMP_STAT"].value_counts()

S    1281549
F     464929
T     244161
U     132299
P       9180
O       4677
R        251
N         66
0         30
Name: EMP_STAT, dtype: int64

In [21]:
obj_df["MARITAL_STAT"].value_counts()

M    1740858
S     175009
E     168292
P      33513
D      17541
0       1420
R        509
Name: MARITAL_STAT, dtype: int64

In [22]:
obj_df["SPOUSE_EMP_STAT"].value_counts()

0    1967401
O     109637
S      36787
F      17363
U       3213
T       2590
P        138
R         13
Name: SPOUSE_EMP_STAT, dtype: int64

In [23]:
pd.get_dummies(obj_df, columns=["MARITAL_STAT", "EMP_STAT","SPOUSE_EMP_STAT","PRED_RISK_TYPE"], prefix=["MARITAL_", "drive","SPOUSE_EMP_","PRED_RISK_TYPE"]).head()

CUSTOMER_NO HOME_BRN RECORD_STAT DATA_EXT_DT DATA_COLL_FROM_DT  \
0   000000418        0           O   15-JUL-19         14-JAN-18   
1   000000422        0           O   15-JUL-19         14-JAN-18   
2   000000430        0           O   15-JUL-19         14-JAN-18   
3   000000433        0           O   15-JUL-19         14-JAN-18   
4   000000437        0           O   15-JUL-19         14-JAN-18   

  DATA_COLL_TO_DT CHURN_STAT_DT HIST_MOVE_DATE  MARITAL__0  MARITAL__D  ...  \
0       14-JAN-19     14-JAN-19      21-NOV-19           0           0  ...   
1       14-JAN-19     14-JAN-19      21-NOV-19           0           0  ...   
2       14-JAN-19     14-JAN-19      21-NOV-19           0           0  ...   
3       14-JAN-19     14-JAN-19      21-NOV-19           0           0  ...   
4       14-JAN-19     14-JAN-19      21-NOV-19           0           0  ...   

   SPOUSE_EMP__R  SPOUSE_EMP__S  SPOUSE_EMP__T  SPOUSE_EMP__U  \
0              0              0              0              0   
1              0              0              0              0   
2              0              0              0              0   
3              0              0              0              0   
4              0              0              0              0   

   PRED_RISK_TYPE_0  PRED_RISK_TYPE_HIGH  PRED_RISK_TYPE_LOW  \
0                 0                    0                   0   
1                 0                    0                   0   
2                 0                    0                   0   
3                 0                    1                   0   
4                 0                    0                   0   

   PRED_RISK_TYPE_MEDIUM  PRED_RISK_TYPE_NO RISK  PRED_RISK_TYPE_TRANSITION  
0                      0                       1                          0  
1                      0                       1                          0  
2                      0                       1                          0  
3                      0                       0                          0  
4                      0                       1                          0  

[5 rows x 38 columns]

In [24]:
obj_df["RECORD_STAT"].value_counts()

O    2126807
C      10335
Name: RECORD_STAT, dtype: int64

In [25]:
obj_df["PRED_RISK_TYPE"].value_counts()

0             1690561
NO RISK        252566
HIGH           156687
MEDIUM          28564
LOW              6419
TRANSITION       2345
Name: PRED_RISK_TYPE, dtype: int64

In [26]:
data_processed = pd.get_dummies(df, columns=["MARITAL_STAT", "EMP_STAT","SPOUSE_EMP_STAT","PRED_RISK_TYPE"], prefix=["MARITAL_", "drive","SPOUSE_EMP_","PRED_RISK_TYPE"])

In [27]:
#record status-0/1 value 


In [28]:
from sklearn.preprocessing import LabelEncoder

lb_make = LabelEncoder()
data_processed["RECORD_STAT"] = lb_make.fit_transform(data_processed["RECORD_STAT"])
#obj_df[["make", "make_code"]].head(11)
data_processed

CUSTOMER_NO HOME_BRN  REP_CCY  CUST_AGE  NUM_DEPEND  CUST_INC  \
0         000000418        0      0.0        44           1     375.0   
1         000000422        0      0.0        39           2     506.0   
2         000000430        0      0.0        30           0       0.0   
3         000000433        0      0.0        77           0       0.0   
4         000000437        0      0.0        37           3     495.5   
...             ...      ...      ...       ...         ...       ...   
2137137   133146421      133      0.0        37           3       0.0   
2137138   091166504       91      0.0        34           1     200.0   
2137139   201166506      201      0.0        39           3     304.0   
2137140   000146441        0      0.0        22           0     145.0   
2137141   141166630      141      0.0        52           9       0.0   

         OVERALL_LIMIT  LIMIT_UTIL_PER  REL_PERIOD  ACC_OPEN_CLOS_RATIO  ...  \
0                  0.0             0.0        1371                0.000  ...   
1                  0.0             0.0        1371                0.500  ...   
2                  0.0             0.0        1371                0.667  ...   
3                  0.0             0.0        1371                0.500  ...   
4                  0.0             0.0        1371                0.333  ...   
...                ...             ...         ...                  ...  ...   
2137137            0.0             0.0          16                0.000  ...   
2137138            0.0             0.0          16                0.000  ...   
2137139            0.0             0.0          16                0.000  ...   
2137140            0.0             0.0          16                0.000  ...   
2137141            0.0             0.0          13                0.000  ...   

         SPOUSE_EMP__R  SPOUSE_EMP__S  SPOUSE_EMP__T  SPOUSE_EMP__U  \
0                    0              0              0              0   
1                    0              0              0              0   
2                    0              0              0              0   
3                    0              0              0              0   
4                    0              0              0              0   
...                ...            ...            ...            ...   
2137137              0              0              0              0   
2137138              0              0              0              0   
2137139              0              0              0              0   
2137140              0              0              0              0   
2137141              0              0              0              0   

         PRED_RISK_TYPE_0  PRED_RISK_TYPE_HIGH  PRED_RISK_TYPE_LOW  \
0                       0                    0                   0   
1                       0                    0                   0   
2                       0                    0                   0   
3                       0                    1                   0   
4                       0                    0                   0   
...                   ...                  ...                 ...   
2137137                 1                    0                   0   
2137138                 1                    0                   0   
2137139                 1                    0                   0   
2137140                 1                    0                   0   
2137141                 1                    0                   0   

         PRED_RISK_TYPE_MEDIUM  PRED_RISK_TYPE_NO RISK  \
0                            0                       1   
1                            0                       1   
2                            0                       1   
3                            0                       0   
4                            0                       1   
...                        ...                     ...   
2137137                      0                       0   
2137138     

In [29]:
#Disply the result
data_processed

CUSTOMER_NO HOME_BRN  REP_CCY  CUST_AGE  NUM_DEPEND  CUST_INC  \
0         000000418        0      0.0        44           1     375.0   
1         000000422        0      0.0        39           2     506.0   
2         000000430        0      0.0        30           0       0.0   
3         000000433        0      0.0        77           0       0.0   
4         000000437        0      0.0        37           3     495.5   
...             ...      ...      ...       ...         ...       ...   
2137137   133146421      133      0.0        37           3       0.0   
2137138   091166504       91      0.0        34           1     200.0   
2137139   201166506      201      0.0        39           3     304.0   
2137140   000146441        0      0.0        22           0     145.0   
2137141   141166630      141      0.0        52           9       0.0   

         OVERALL_LIMIT  LIMIT_UTIL_PER  REL_PERIOD  ACC_OPEN_CLOS_RATIO  ...  \
0                  0.0             0.0        1371                0.000  ...   
1                  0.0             0.0        1371                0.500  ...   
2                  0.0             0.0        1371                0.667  ...   
3                  0.0             0.0        1371                0.500  ...   
4                  0.0             0.0        1371                0.333  ...   
...                ...             ...         ...                  ...  ...   
2137137            0.0             0.0          16                0.000  ...   
2137138            0.0             0.0          16                0.000  ...   
2137139            0.0             0.0          16                0.000  ...   
2137140            0.0             0.0          16                0.000  ...   
2137141            0.0             0.0          13                0.000  ...   

         SPOUSE_EMP__R  SPOUSE_EMP__S  SPOUSE_EMP__T  SPOUSE_EMP__U  \
0                    0              0              0              0   
1                    0              0              0              0   
2                    0              0              0              0   
3                    0              0              0              0   
4                    0              0              0              0   
...                ...            ...            ...            ...   
2137137              0              0              0              0   
2137138              0              0              0              0   
2137139              0              0              0              0   
2137140              0              0              0              0   
2137141              0              0              0              0   

         PRED_RISK_TYPE_0  PRED_RISK_TYPE_HIGH  PRED_RISK_TYPE_LOW  \
0                       0                    0                   0   
1                       0                    0                   0   
2                       0                    0                   0   
3                       0                    1                   0   
4                       0                    0                   0   
...                   ...                  ...                 ...   
2137137                 1                    0                   0   
2137138                 1                    0                   0   
2137139                 1                    0                   0   
2137140                 1                    0                   0   
2137141                 1                    0                   0   

         PRED_RISK_TYPE_MEDIUM  PRED_RISK_TYPE_NO RISK  \
0                            0                       1   
1                            0                       1   
2                            0                       1   
3                            0                       0   
4                            0                       1   
...                        ...                     ...   
2137137                      0                       0   
2137138     

In [30]:
data_processed.shape

(2137142, 100)

In [31]:
# pandas drop columns using list of column names
data_processed= data_processed.drop(['DATA_EXT_DT', 'DATA_COLL_FROM_DT', 'DATA_COLL_TO_DT','CHURN_STAT_DT','HIST_MOVE_DATE'], axis=1)

In [32]:
data_processed1=data_processed.drop(['CUSTOMER_NO','HOME_BRN'],axis=1)

In [33]:
data_processed1

REP_CCY  CUST_AGE  NUM_DEPEND  CUST_INC  OVERALL_LIMIT  \
0            0.0        44           1     375.0            0.0   
1            0.0        39           2     506.0            0.0   
2            0.0        30           0       0.0            0.0   
3            0.0        77           0       0.0            0.0   
4            0.0        37           3     495.5            0.0   
...          ...       ...         ...       ...            ...   
2137137      0.0        37           3       0.0            0.0   
2137138      0.0        34           1     200.0            0.0   
2137139      0.0        39           3     304.0            0.0   
2137140      0.0        22           0     145.0            0.0   
2137141      0.0        52           9       0.0            0.0   

         LIMIT_UTIL_PER  REL_PERIOD  ACC_OPEN_CLOS_RATIO  LN_DP_AMT_RATIO  \
0                   0.0        1371                0.000              0.0   
1                   0.0        1371                0.500              0.0   
2                   0.0        1371                0.667              0.0   
3                   0.0        1371                0.500              0.0   
4                   0.0        1371                0.333              0.0   
...                 ...         ...                  ...              ...   
2137137             0.0          16                0.000              0.0   
2137138             0.0          16                0.000              0.0   
2137139             0.0          16                0.000              0.0   
2137140             0.0          16                0.000              0.0   
2137141             0.0          13                0.000              0.0   

         LN_DP_AC_RATIO  ...  SPOUSE_EMP__R  SPOUSE_EMP__S  SPOUSE_EMP__T  \
0                  2.00  ...              0              0              0   
1                  2.00  ...              0              0              0   
2                  1.00  ...              0              0              0   
3                  0.00  ...              0              0              0   
4                  1.33  ...              0              0              0   
...                 ...  ...            ...            ...            ...   
2137137            0.00  ...              0              0              0   
2137138            0.00  ...              0              0              0   
2137139            0.00  ...              0              0              0   
2137140            0.00  ...              0              0              0   
2137141            0.00  ...              0              0              0   

         SPOUSE_EMP__U  PRED_RISK_TYPE_0  PRED_RISK_TYPE_HIGH  \
0                    0                 0                    0   
1                    0                 0                    0   
2                    0                 0                    0   
3                    0                 0                    1   
4                    0                 0                    0   
...                ...               ...                  ...   
2137137              0                 1                    0   
2137138              0                 1                    0   
2137139              0                 1                    0   
2137140              0                 1                    0   
2137141              0                 1                    0   

         PRED_RISK_TYPE_LOW  PRED_RISK_TYPE_MEDIUM  PRED_RISK_TYPE_NO RISK  \
0                         0                      0                       1   
1                         0                      0                       1   
2                         0                      0                       1   
3                         0                      0                       0   
4                         0                      0                       1   
...                     ...                    ...                     ...   
2137137                   0

In [ ]:
from sklearn.neighbors import NearestNeighbors
import numpy as np
#X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
X=np.array(data_processed1)
nbrs = NearestNeighbors(n_neighbors=4, algorithm='auto').fit(X)
distances, indices = nbrs.kneighbors(X)
indices

In [ ]:
from sklearn.neighbors import KDTree
import numpy as np
X=np.array(data_processed1)
kdt = KDTree(X, leaf_size=50, metric='euclidean')
kdt.query(X, k=2, return_distance=True)

In [ ]:

sklearn.metrics.pairwise.pairwise_distances
pairwise_distances(X, Y=None, metric='euclidean', n_jobs=1, **kwds)

In [37]:
from sklearn.cluster import KMeans
kmeans = KMeans(n_clusters = 10, random_state = 0)
clusters = kmeans.fit_predict(data_processed1)
kmeans.cluster_centers_

array([[ 0.00000000e+00,  4.22740091e+01,  2.46382466e+00,
         2.09301033e+02,  0.00000000e+00,  0.00000000e+00,
         7.47349086e+02,  8.55476480e-03,  6.65122262e-06,
         1.06134173e-03,  6.89943116e-03,  5.96943777e+02,
         1.02372694e-03,  1.35490641e-02,  7.61140444e-01,
         6.78058654e-01,  6.96831612e-01,  6.20095372e-01,
        -3.51647498e+01, -1.13377716e+00,  3.29268493e+00,
        -4.63412368e+01,  6.75404878e-01,  3.40567269e+00,
         6.87979225e+01,  2.86954358e-02,  3.44439696e+00,
        -4.95424264e+01, -1.47443274e+00,  2.93529728e+00,
         2.92233966e-03, -2.41113330e-02,  1.93469443e-01,
        -8.32128401e-01,  3.69763400e-02,  1.88558859e-01,
         3.57861108e-01,  9.57580436e-02,  2.30666287e-01,
        -7.29881585e-03, -4.06200543e-02,  2.29448312e-01,
         0.00000000e+00,  0.00000000e+00,  0.00000000e+00,
         0.00000000e+00,  0.00000000e+00,  0.00000000e+00,
         0.00000000e+00,  0.00000000e+00,  0.00000000e+0

In [39]:
print(kmeans.cluster_centers_.shape)

(10, 93)


In [35]:
scores =kmeans.score(data_processed1) 

In [36]:
scores

-1.8026667021840804e+16